### ISSUE TITLE
`Style/RaiseArgs` vs receiverless `new`

### ISSUE THREAD

**Issue by tsugimoto** [rubocop/rubocop#10506]
I was too dumb to write 

```sh
$ cat main.rb
```

main.rb
```rb
# frozen_string_literal: true

raise new NotImplementedError
```

and it surprised me.

--------

## Expected behavior

This code is syntactically valid Ruby, so `rubocop` should exit normally without errors or warnings.

## Actual behavior

```
$ rubocop --debug
For ...: Default configuration from .../ruby/gems/3.1.0/gems/rubocop-1.26.1/config/default.yml
Use parallel by default.
Running parallel inspection
An error occurred while Style/RaiseArgs cop was inspecting .../main.rb:3:0.
undefined method `const_name' for nil:NilClass

          type = arg.receiver.const_name
                             ^^^^^^^^^^^
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/style/raise_args.rb:138:in `allowed_non_exploded_type?'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/style/raise_args.rb:113:in `check_exploded'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/style/raise_args.rb:60:in `on_send'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:136:in `public_send'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:136:in `block (2 levels) in trigger_restricted_cops'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:135:in `block in trigger_restricted_cops'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:134:in `each'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:134:in `trigger_restricted_cops'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:70:in `on_send'
.../ruby/gems/3.1.0/gems/rubocop-ast-1.16.0/lib/rubocop/ast/traversal.rb:20:in `walk'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:86:in `investigate'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/team.rb:155:in `investigate_partial'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/team.rb:83:in `investigate'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:309:in `inspect_file'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:253:in `block in do_inspection_loop'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:287:in `block in iterate_until_no_changes'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:280:in `loop'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:280:in `iterate_until_no_changes'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:249:in `do_inspection_loop'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:130:in `block in file_offenses'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:155:in `file_offense_cache'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:129:in `file_offenses'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:67:in `block in warm_cache'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:587:in `call_with_index'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:557:in `process_incoming_jobs'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:537:in `block in worker'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:528:in `fork'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:528:in `worker'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:519:in `block in create_workers'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:518:in `each'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:518:in `each_with_index'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:518:in `create_workers'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:457:in `work_in_processes'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:294:in `map'
.../ruby/gems/3.1.0/gems/parallel-1.22.0/lib/parallel.rb:238:in `each'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:67:in `warm_cache'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:46:in `run'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/command.rb:11:in `run'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/environment.rb:18:in `run'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli.rb:71:in `run_command'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli.rb:78:in `execute_runners'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli.rb:47:in `run'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/exe/rubocop:12:in `block in <top (required)>'
.../ruby/3.1.0/benchmark.rb:311:in `realtime'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/exe/rubocop:12:in `<top (required)>'
.../bin/rubocop:25:in `load'
.../bin/rubocop:25:in `<main>'
Inspecting 1 file
Scanning .../main.rb
An error occurred while Style/RaiseArgs cop was inspecting .../main.rb:3:0.
undefined method `const_name' for nil:NilClass

          type = arg.receiver.const_name
                             ^^^^^^^^^^^
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/style/raise_args.rb:138:in `allowed_non_exploded_type?'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/style/raise_args.rb:113:in `check_exploded'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/style/raise_args.rb:60:in `on_send'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:136:in `public_send'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:136:in `block (2 levels) in trigger_restricted_cops'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:135:in `block in trigger_restricted_cops'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:134:in `each'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:134:in `trigger_restricted_cops'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:70:in `on_send'
.../ruby/gems/3.1.0/gems/rubocop-ast-1.16.0/lib/rubocop/ast/traversal.rb:20:in `walk'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/commissioner.rb:86:in `investigate'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/team.rb:155:in `investigate_partial'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cop/team.rb:83:in `investigate'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:309:in `inspect_file'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:253:in `block in do_inspection_loop'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:287:in `block in iterate_until_no_changes'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:280:in `loop'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:280:in `iterate_until_no_changes'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:249:in `do_inspection_loop'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:130:in `block in file_offenses'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:155:in `file_offense_cache'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:129:in `file_offenses'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:120:in `process_file'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:101:in `block in each_inspected_file'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:100:in `each'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:100:in `reduce'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:100:in `each_inspected_file'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:86:in `inspect_files'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/runner.rb:47:in `run'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/command.rb:11:in `run'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli/environment.rb:18:in `run'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli.rb:71:in `run_command'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli.rb:78:in `execute_runners'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/lib/rubocop/cli.rb:47:in `run'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/exe/rubocop:12:in `block in <top (required)>'
.../ruby/3.1.0/benchmark.rb:311:in `realtime'
.../ruby/gems/3.1.0/gems/rubocop-1.26.1/exe/rubocop:12:in `<top (required)>'
.../bin/rubocop:25:in `load'
.../bin/rubocop:25:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/RaiseArgs cop was inspecting .../main.rb:3:0.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.26.1 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 3.1.1 x86_64-darwin20)
Finished in 0.4795440000016242 seconds
$ 
```

## Steps to reproduce the problem

Commands are described in above code.

## RuboCop version

```
$ rubocop -V
1.26.1 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 3.1.1 x86_64-darwin20)
$ 
```
